### DISCUSSION TITLE
Direnv

### DISCUSSION THREAD

**Discussion by sitolam** [NvChad/NvChad#2916] [Q&A]
How can I use the direnv nvim plugin with nvchad?

**Comment by siduck** [comment]
how would you use it with other neovim configs? please read our nvchad.com plugin docs to know how to install a plugin

**Reply by SamoylovMD** [reply]
I +1 to this question. Disclaimer: I have a big monorepo with many subprojects having their own .env/.envrc configs. I also have `vim.opt.autochdir = true`, so my direnv is reloaded everytime when current directory changes.

I installed direnv.vim to my nvchad, and it kinda works - but I think it sometimes clashes with tabufline. I didn't find any description how this tab plugin works exactly - and sometimes when working with tabs intensively, I find direnv integration broken, throwing errors every time when I change a tab or open a new file. 

Ideally, I would expect direnv to reload every time I change tab (because it means that current directory should also change), but I cannot find a proper event to attach autocmd - BufEnter didn't seem to work correctly. 

So, my more specific question is - is there a way to reload direnv on each tab open/switch? Is there any kind of event emitted by tabufline when I switch tabs?

**Reply by siduck** [reply]
show those errors, make an issue on ui repo with steps so i can reproduce and fix
